active/gridsyntax/GridSyntaxSpec.md

```diff
@@ -0,0 +1,335 @@
+          <RowDefinition Height="1*" />
+          <RowDefinition Height="Auto" />
+          <RowDefinition Height="25" />
+          <RowDefinition Height= "14" />
+          <RowDefinition Height="20" />
+    </Grid.RowDefinitions>
+</Grid>
+```
+
+This code creates multiple ColumnDefinition objects and RowDefinitions objects and adds them to the Grid's ColumnDefinitions and RowDefinitions collections, respectively. ColumnDefinitions and RowDefinitions are read-only properties of Grid, and the syntax shown above is currently the only way to update these properties. Due to their read-only nature, existing typeconverters cannot be used with these properties. 
+
+The existing syntax supports specialized functionality such as:
+- Declaring max/min height or width
+```xml
+<Grid MaxHeight="600" MaxWidth="600">
+  ...
+</Grid>
+```
+
+- Data binding for row/height values:
+```xml
+<Grid>
+  <Grid.RowDefinitions>
+    <RowDefinition Height="{x:Bind sampleDataValue}" />
+    ...
+  </Grid.RowDefinitions>
+</Grid>
+```
+Grid's usability, intuitiveness, and developer satsifaction will be improved upon by utilizing a more succinct syntax, as will be described throughout this spec.
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+<!-- This section and the appendix are the only sections that likely
+do not get copied to docs.microsoft.com; they're just an aid to reading this spec. -->
+
+<!-- If you're modifying an existing API, included a link here to the
+existing page(s) -->
+
+<!-- For example, this section is a place to explain why you're adding this API rather than
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+# Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+## Overview
+A new XAML language feature that allows the initialization of collection-type properties (even if they are read-only) using element attribute syntax.
+
+## Implementation Details
+Read-only collection-type properties will be assigned a new CreateFromString attribute. The CreateFromString attribute will allow them to be defined/created from a comma-delimited string. In terms of Grid, this means that the ColumnDefinitions and RowDefinitions properties are able to be defined by providing a comma-delimited string of ColumnDefinitions (Width values) and a comma-delimited string of RowDefinitions (Height values), respectively. However, this feature is also applicable to other scenarios within the XAML language where collection-typed read-only properties are created (see Remarks for examples of such scenarios). 
```

Comment:
At a high level, what we're implementing is the following:
1. When the parser encounters an element attribute that is assigning to a collection-type property, it recognizes that initialization of a collection from a string is being requested.
2. The parser tokenizes the attribute value (a comma-delimited string) into a series of substrings, each of which represents the initialization value of a single element of the collection
3. The parser will perform the usual "initialize an instance of a type from a string" operation. 
4. If the type has a ContentProperty, then a new instance of the type will be created (zero-argument constructor) and the string value will be assigned to the content property (this might in turn invoke a type converter, but the details are at a lower abstraction level and so are not germane to this discussion).
5. If the type has either a built-in type converter or a CreateFromString attribute specifying a factory method, then that will be invoked with the string value as an argument, returning a new instance of the type.
6. If neither step 4 nor 5 is performed, then this is an error.

For Grid-specific portion of this work, we'll be following step 4 from above, adding the ContentProperty attribute to RowDefinition and ColumnDefinition.

Reply:
Hope its okay that I used your words in the spec - added this in.